File: services/asr-service/asr-server/src/main/java/com/livetranslate/asr/domain/model/ASRSession.java
package com.livetranslate.asr.domain.model;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * ASR 会话聚合根
 * 对应原 TypeScript 的 connState
 */
@Data
public class ASRSession {

    /** 会话 ID */
    private String sessionId;

    /** API Key */
    private String apiKey;

    /** 源语言 */
    private String sourceLang;

    /** 目标语言 */
    private String targetLang;

    /** ASR 模型 */
    private String model;

    /** 采样率 */
    private Integer sampleRate;

    /** 会话状态 */
    private SessionStatus status;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 最后活跃时间 */
    private LocalDateTime lastActiveAt;

    public enum SessionStatus {
        INITIALIZING,
        CONNECTING,
        READY,
        PROCESSING,
        CLOSED
    }
}

File: services/asr-service/asr-server/src/domain/model/ASRSession.ts
/**
 * ASR 会话领域模型
 */
export interface ASRSession {
  sessionId: string;
  apiKey: string;
  sourceLang: string;
  targetLang: string;
  model: string;
  status: 'initializing' | 'connecting' | 'ready' | 'processing' | 'closed';
  createdAt: Date;
  lastActiveAt: Date;
}
